lib/internal/navigator.js

```diff
@@ -23,7 +25,7 @@ const nodeVersion = process.version;
 class Navigator {
   // Private properties are used to avoid brand validations.
   #availableParallelism;
-  #userAgent = `Node.js/${nodeVersion.slice(1, nodeVersion.indexOf('.'))}`;
+  #userAgent = `Node.js/${StringPrototypeSlice(nodeVersion, 1, StringPrototypeIndexOf(nodeVersion, '.'))}`;
```

Comment:
What's the reason for using primordials in here? What's the benefit, and what's the caveat for not using it?

Reply:
Our onboarding document states that:

Insofar as possible, issues should be identified by tools rather than human reviewers. If you are leaving comments about issues that could be identified by tools but are not, consider implementing the necessary tooling.

This is an issue that can be identified by tools rather than human reviewers. If we expect this change to pass, we should implement the necessary tooling.

Reply:
I don’t see how; there’s no way to know the type of the thing and thus whether it’s required except by a human.

Reply:
I guess the only issue is, that if somebody instantiates for sOmE reason a Navigator instance AND String.prototype.slice or .indexOf is monkey patched, that something unexpected could happen...

Reply:
The result will always start with Node.js prefix. What unexpected issue might occur by tampering with string.prototype.slice. I think we're optimizing against a hypotethical situation, and this is the reason that got us behind the current situation that led to primordials strategic initiative.

Reply:
Until that initiative comes to a conclusion tho, nothing should change.

The problem is that nothing unexpected is supposed to happen by tampering with builtins, which is the current status quo - not an exception or the wrong result, especially if we’re trying to implement web platform features which have this property.

Reply:
What @anonrig is saying, and I agree - is that weird things **should** be expected if you tamper `String.prototype`

Reply:
>  this is the reason that got us behind the current situation that led to primordials strategic initiative.

FWIW unfortunately I will only have time to re-engage with the initiative at a non-intermittent capacity after the war is over.

Reply:
> What’s the reason for using primordials in here? What’s the benefit, and what’s the caveat for not using it?

While I agree with @anonrig that using primordials here is pointless, I think it’s required by our policy in https://github.com/nodejs/node/blob/main/doc/contributing/primordials.md. I expect that policy to change once @nodejs/primordials gets a bit further along in their work, but until they do, I can’t really fault @Uzlopak for following the rules as currently written.

Reply:
I don't think it's required by our policy. The wording of the second paragraph states that it is **preferred** for any new code. Additionally, the latest [TSC voting states](https://github.com/nodejs/TSC/blob/main/votes/2022-02-21-0.json):

```
"Use primordials in the error path only if there are no significant perf regression."
```

This is not an error path.

Reply:
The vote only refers to code in an error path, so indeed that decision doesn't apply here. "Preferred" may indeed be the case.

However, this is a Web Platform standard feature, and to comply with web standards, all web platform features need to be robust in the exact ways primordials ensure - what's the point of implementing web standards only to not fully follow them?

Reply:
This is a private property on a global singleton object. Is it even possible to mutate string.prototype before Navigator is initialized?

Reply:
That depends on whether user code can run first. I completely agree that if it’s a singleton, the constructor is never available to user code, and it instantiates before any user code can run (including loaders or —require or —import) that this file wouldn’t need to use them in fields.

However, that’s a lot of caveats - and if it’s only invoked once than how bad could the perf hit really be anyways?